odoo/models.py

```diff
@@ -1741,19 +1790,362 @@ def clear_caches(cls):
         """
         cls.pool._clear_cache()
 
+    @api.model
+    def _read_group(self, domain, groupby=(), aggregates=(), having=(), offset=0, limit=None, order=None):
+        """ Get fields aggregations specified by ``aggregates`` grouped by the given ``groupby``
+        fields where record are filtered by the ``domain``.
+
+        :param list domain: :ref:`A search domain <reference/orm/domains>`. Use an empty
+                list to match all records.
+        :param list groupby: list of groupby descriptions by which the records will be grouped.
+                A groupby description is either a field (then it will be grouped by that field)
+                or a string 'field:granularity'. Right now, the only supported granularities
+                are 'day', 'week', 'month', 'quarter' or 'year', and they only make sense for
+                date/datetime fields.
+        :param list aggregates: list of aggregates specification.
+                Each element is 'field:agg' (aggregate field with aggregation function 'agg').
+                The possible aggregation functions are the ones provided by
+                `PostgreSQL <https://www.postgresql.org/docs/current/static/functions-aggregate.html>`_,
+                'count_distinct' with the expected meaning and 'recordset' to act like 'array_agg'
+                converted into a recordset.
+        :param list having: A domain where the valid "fields" are the aggregates.
+        :param int offset: optional number of groups to skip
+        :param int limit: optional max number of groups to return
+        :param str order: optional ``order by`` specification, for
+                overriding the natural sort ordering of the groups,
+                see also :py:meth:`~osv.osv.osv.search`.
```

Comment:
I assume you meant to create a link to the method here?
```suggestion
                see also :meth:`~.search`.
```